From: Egbert Eich <eich@suse.de>
Date: Tue Jan 28 13:20:10 2014 +0100
Subject: drm/mgag200: Free container instead of member in mga_user_framebuffer_destroy()
Git-repo: git://people.freedesktop.org/~airlied/linux/ mgag200-queue
Git-commit: 798fc9d0cedfacf3fed805293f57a80fd30b9b82
References: bnc#806990
Patch-Mainline: to be upstreamed
Signed-off-by: Egbert Eich <eich@suse.com>

Technically freeing mga_fb->base is the same as freeing mga_fb as 'base'
the first member of the data structure.
Still this makes it cleaner.

Signed-off-by: Egbert Eich <eich@suse.de>
---
 drivers/gpu/drm/mgag200/mgag200_main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/mgag200/mgag200_main.c b/drivers/gpu/drm/mgag200/mgag200_main.c
index 0f8b861..76cd683 100644
--- a/drivers/gpu/drm/mgag200/mgag200_main.c
+++ b/drivers/gpu/drm/mgag200/mgag200_main.c
@@ -20,7 +20,7 @@ static void mga_user_framebuffer_destroy(struct drm_framebuffer *fb)
 	if (mga_fb->obj)
 		drm_gem_object_unreference_unlocked(mga_fb->obj);
 	drm_framebuffer_cleanup(fb);
-	kfree(fb);
+	kfree(mga_fb);
 }
 
 static const struct drm_framebuffer_funcs mga_fb_funcs = {
